bibliopixel/remote/control.py

```diff
@@ -14,89 +13,95 @@
     'valid': True
 }
 
-DEFAULT_SERVER_CONFIG = {
-    'title': 'BiblioPixel Remote',
-    'bgcolor': '#000000',
-    'font_color': '#ffffff',
-    'external_access': False,
-    'port': 5000
+
+OFF_ANIM = {
+    'animation': 'bibliopixel.animation.off.OffAnim',
+    'run': {'threaded': True}
 }
 
+
 BAD_DEFAULT_ERROR = """\
 `{}` is not a valid default!
 It must be one of the configured animation names."""
 
 
-class RemoteControl:
-    def __init__(self, config, animations):
-        # These start with the defaults
-        self.config = dict(DEFAULT_SERVER_CONFIG, **config)
+class RemoteControl(collection.Collection):
+    def __init__(self, layout, animations,
+                 external_access=False, port=5000,
+                 title='BiblioPixel Remote', bgcolor='black',
+                 font_color='white', default=None):
+
+        self.internal_delay = 0  # never wait
+
+        # normalize name and display first
+        for anim in animations:
+            anim['animation']['data'] = dict(DEFAULT_ANIM_CONFIG, **anim['animation'].get('data', {}))
+            if not anim['animation'].get('name'):
+                raise ValueError('All animations require the `name` parameter: {}'.format(anim['animation']))
+            if not anim['animation']['data'].get('display'):
```

Comment:
Ahhh... nope, that's no good. Because .get() only chooses your default option if that key DOESN'T EXIST. But in this it does because I always defaulted it to None. In the interest of leaving out the if statement I'm going to instead remove my default to None.